### PR TITLE
[WIP] Implementation for Jandy Watercolors Lights

### PR DESCRIPTION
**Pull request by carlos-sarmiento**
# What does this implement/fix?
Implement support for Jandy Watercolors Lights using a Relay. Manual is here: https://www.jandy.com/en/products/pool-lights/watercolors-led-gen-2

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [ ] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/jandy_watercolors/jandy_watercolors.cpp [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+}
+
+void JandyWatercolorsLightOutput::target_next_effect() {
+  auto new_target_effect = this->target_effect_ + 1;
+  if (new_target_effect >= TOTAL_SUPPORTED_EFFECTS || new_target_effect < JANDY_FIRST_EFFECT) {
+    new_target_effect = JANDY_FIRST_EFFECT;
+  }
+
+  auto call = this->light_state_->turn_on();
+  call.set_effect(this->get_effect_name_from_id_(new_target_effect));
```
Comment (ssieb) [inline esphome/components/jandy_watercolors/jandy_watercolors.cpp:79]: You can give `set_effect` the effect number as well, so you don't need the lookup.

File: esphome/components/jandy_watercolors/jandy_watercolors.cpp [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+}
+
+void JandyWatercolorsLightOutput::update_active_effect_() {
+  if (this->is_light_on_ && this->current_effect_ == this->target_effect_ &&
+      this->target_effect_ >= JANDY_FIRST_EFFECT &&
+      this->light_state_->get_effect_name() != this->get_effect_name_from_id_(this->current_effect_)) {
+    ESP_LOGD(TAG, "Updating Active Effect on UI to: %s", this->get_effect_name_from_id_(this->current_effect_).c_str());
+
+    auto call = this->light_state_->turn_on();
+    call.set_effect(this->get_effect_name_from_id_(this->current_effect_));
```
Comment (ssieb) [inline esphome/components/jandy_watercolors/jandy_watercolors.cpp:164]: Same thing here.

File: esphome/components/jandy_watercolors/jandy_watercolors.h [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  int target_effect_after_reset_ = JANDY_RESET_EFFECT;
+  bool is_output_on_ = false;
+  bool is_light_on_ = false;
+  uint32_t last_turned_off_ = 0;
+  uint32_t last_turned_on_ = 0;
+
+  std::string effect_names_[TOTAL_SUPPORTED_EFFECTS] = {
+      "Alpine White", "Sky Blue",  "Cobalt Blue", "Caribbean Blue",    "Spring Green",      "Emerald Green",
+      "Emerald Rose", "Magenta",   "Violet",      "Slow Color Splash", "Fast Color Splash", "America the Beautiful",
+      "Fat Tuesday",  "Disco Tech"};
```
Comment (ssieb) [inline esphome/components/jandy_watercolors/jandy_watercolors.h:69]: This should be a static array in the .cpp file.

File: esphome/components/jandy_watercolors/jandy_watercolors.h [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+#include "esphome/core/automation.h"
+
+#include "esphome/components/output/binary_output.h"
+#include "esphome/components/light/light_output.h"
+#include "esphome/components/light/light_state.h"
+#include "esphome/components/light/light_effect.h"
+
+namespace esphome {
+namespace jandy_watercolors {
+static const char *const TAG = "jandy-watercolors";
```
Comment (ssieb) [inline esphome/components/jandy_watercolors/jandy_watercolors.h:14]: This needs to be in the .cpp file.

**Review by ssieb — CHANGES_REQUESTED** [review]

**Comment by jesserockz** [comment]
@carlos-sarmiento you marked this as ready to review, but you did not address any of the above feedback.

**Comment by carlos-sarmiento** [comment]
Sorry. Misclicked the button. I asked a few questions on the feedback so I
can implement the changes

**Comment by jesserockz** [comment]
@carlos-sarmiento I think you didnt submit your answers to the review. They are probably still "pending" as I do not see them

**Comment by carlos-sarmiento** [comment]
I was asking for what the syntax for the static array would be on one of @ssieb comments

**Comment by ssieb** [comment]
> I was asking for what the syntax for the static array would be on one of @ssieb comments

Just put `static` in front of it and move it to the `.cpp` file.

**Pull request closed without merging**